cpp/include/cugraph/algorithms.hpp

```diff
@@ -1344,6 +1344,8 @@ extract_ego(raft::handle_t const& handle,
  * (compressed) format; when padding is used the output is a matrix of vertex paths and a matrix of
  * edges paths (weights); in this case the matrices are stored in row major order; the vertex path
  * matrix is padded with `num_vertices` values and the weight matrix is padded with `0` values;
+ * @param selector_type identifier for sampling strategy: uniform, biased, etc.; defaults to
```

Comment:
Should this be an enum?  We can have an enum for the public API that is separate from the internal (detail) enum if there's a reason not to expose the internal details.

If we're going to use an int, should the documentation identify the integer values that correspond to the current supported types.

Reply:
I don't mind an `enum` but that enum would reside where? What header? We cannot place it in the implementation headers, because that would create circular header inclusions. It's also probably not that helpful to create a separate header only for that enum. 

I don't see any other such headers being included in `algorithms.hpp`. Also, there's only one other `enum` being defined in `algorithms.hpp`. And that is a scoped `enum`, which must be type-erased to an `int` anyway for either C-only API or Python. So, why not use an `int` directly? (at least in the public API that other languages must see and use)

(And, yes I'll add dox about the one other possible value for sampler selection, which is `1`. The default, `0` is already docummented).

Reply:
@aschaffer I agree that picking a place for the enum to be defined is an issue.  Maybe just define the int values in the docs and push enum definition to the python side for now.  We should have a later discussion on enums since this is not the only instance

Reply:
Done (documented better the integer values).

Reply:
Resolving this for now.

Reply:
Generally, the algorithms exposed in `algorithms.hpp` don't take enumerated types that control the behavior of the algorithm.  The only one that I know of (before this) is the legacy connected components implementation, which defines the enumerated type in this file.  Note that the new weakly connected components implementation (the MNMG version) is a separate entry point.  It is unclear, at this time, if the MNMG strongly connected component implementation will be a separate entry point or add the enumerated type to the weakly connected components implementation.

Other places where this might be appropriate we create separate entry points in the `algorithms.hpp` and reuse the code in the implementation.  For example, the spectral clustering algorithms are each exposed as separate entry points.  They could be exposed as a single entry point with an enumerated type identifying which variation to run.  I don't know what motivated the choice between approaches... probably left to the preference of the developer creating the algorithms.

I'm OK with just using an int as long as we define it in the documentation.

Reply:
+1 for Chuck's idea to "create separate entry points in the algorithms.hpp and reuse the code in the implementation'', but yeah... just a recommendation.

Reply:
I'm not so sure about that. Unless I'm misunderstanding what's proposed here, when you have many moving parts (i.e., policies) the number of "separate entry points" would explode. For example, if we were to expose RW with (not even) all its moving parts: `random_engine_t` (let's say you have R possible choices), `traversal_t` (T possible choices), `sampling_t` (S possible choices); you would have to add function names for each combination, resulting in R*T*S "separate entry points". 

This is moving the named choice...into the name of the function. You're essentially unwrapping function instantiations (in the case of template functions, which is our case). Which really is compiler's job, not programmer's.

Reply:
Moreover, imagine over time you add one or more policies. Say, `neighborhood_discovery_t`. You must now go back and change the names of all the previous separate entry points, as opposed to just adding an additional `enum` or `int` argument. And while an additional argument might not break existent client code (if, for example, you make that argument optional); changing names on existent API certainly would (break caller's code).